### ISSUE TITLE
Visual Studio Test results are not published if build timeouts

### ISSUE THREAD

**Issue by LokiMidgard** [Microsoft/vsts-tasks#7701]
## Environment
TFS 2018 with private agent version 2.134.2 on Windows 7

## Issue Description

When a Build timeouts and stops a running Visual Studio Test, all already finished test results are lost. I assume Test results are published at the end of the Visual Studio Test, which is not reached.

**Comment by abhishkk** [comment]
@LokiMidgard 
In case of single agent scenario, we run all the tests in one go on single agent. If build timeout happens, trx file is not generated and thus test results are not published.

In case of multi-agent scenario, we distribute the tests into slices and pass it to multiple agents. If any agent is done with test execution before build timeout, you will be able to see the test results published for them.

We are also actively working on to improve this experience of publishing test results.

**Comment by LokiMidgard** [comment]
Thank you for your reply. I'm looking forward to it.

While many things are better in TFS 2018 then TFS 2010, the fact that I saw live how many tests are passed/faild and not jet executed, was a great help when testing bigger changes in TFS2010. If after 10 minutes there ware already errors, I did not need to waste 3 or more hours to start investigating. Especially aborting a test run that already had several errors were possible without losing the data.
